File: 153.find-minimum-in-rotated-sorted-array.js
// @leet start
/**
 * @param {number[]} nums
 * @return {number}
 */
var findMin = function(nums) {
	if (nums.length === 1) {
		return nums[0];
	}
	var left = 0;
	var right = nums.length - 1;
	while (left < right) {
		var middle = Math.floor((right - left) / 2 + left);
		if (nums[middle] < nums[right]) {
			right = middle;
		} else {
			left = middle+1;
		}
	}
	return nums[left];
};
// @leet end

File: 153.find-minimum-in-rotated-sorted-array.cpp
// @leet start
#include <vector>
using namespace std;
class Solution {
public:
  int findMin(vector<int> &nums) {
		int left=0;
		int right=nums.size()-1;
		while (left<right) {
			if (nums[left + (right-left)/2] <nums[right]) {
				right=left + (right-left)/2;
			}
			else {
				left=left + (right-left)/2 +1;	
			}
		}
		return nums[left];
  }
};
// @leet end
